cmd/monitorCodeGen/main.go

```diff
@@ -0,0 +1,195 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/signalfx/signalfx-agent/internal/monitors"
+	log "github.com/sirupsen/logrus"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"text/template"
+)
+
+var tmpl = `// Code generated by monitor-code-gen. DO NOT EDIT.
+
+package {{.goPackage}}
+
+import "github.com/signalfx/signalfx-agent/internal/monitors"
+
+{{with .groupMetricsMap}}
+const (
+{{- range . }}
+	{{(printf "group.%s" .Name) | formatVariable}} = "{{.Name}}"
+{{- end}}
+)
+{{end}}
+
+var groupSet = map[string]bool {
+{{- range .groupMetricsMap}}
+	{{(printf "group.%s" .Name) | formatVariable}}: true,
+{{- end}}
+}
+
+{{with .metrics}}
+const (
+{{- range .}}
+	{{.Name | formatVariable}} = "{{.Name}}"
+{{- with .Alias}}
+	{{. | formatVariable}} = "{{.}}"
+{{- end}}
+{{- end}}
+)
+{{end}}
+
+var metricSet = map[string]bool {
+{{- range .metrics}}
+	{{.Name | formatVariable}}: true,
+{{- end}}
+}
+
+var includedMetrics = map[string]bool {
+{{- range .metrics}}
+{{- if .Included}}
+	{{.Name | formatVariable}}: true,
+{{- with .Alias}}
+	{{. | formatVariable}}: true,
+{{- end}}
+{{- end}}
+{{- end}}
+}
+
+var groupMetricsMap = map[string][]string {
+{{- range $group, $metrics := .groupMetricsMap}}
+	{{(printf "group.%s" $group) | formatVariable}}: []string {
```

Comment:
Why does it have a `group.` prefix?

Reply:
If you have group named "mem" it'd result in a variable mem which seems weird and potentially conflicting with local variables. Metric tend to be fairly unique so not as much of a concern there.

Reply:
What am I missing here?  This is generating map key strings which aren't variables and also shouldn't they be put in quotes?  Or does printf/formatVariable add double quote literals around them?

Reply:
Oh I think I misunderstood the original question. So a group "mem" gets a const generated called `groupMem` which is equal to `"mem"`. Then when generating the map it uses the const `groupMem` as the key. Should compile to the equivalent, it's just using our own constants as a safety measure.

Reply:
Here's what it looks like:

```go
var groupMetricsMap = map[string][]string {
	groupContainers: []string {
		kubernetesContainerReady,
		kubernetesContainerRestartCount,
	},
	groupDaemonSets: []string {
		kubernetesDaemonSetCurrentScheduled,
		kubernetesDaemonSetDesiredScheduled,
		kubernetesDaemonSetMisscheduled,
		kubernetesDaemonSetReady,
	},
	groupDeployments: []string {
		kubernetesDeploymentAvailable,
		kubernetesDeploymentDesired,
	},
	groupNamespaces: []string {
		kubernetesNamespacePhase,
	},
	groupNodes: []string {
		kubernetesNodeReady,
	},
	groupPods: []string {
		kubernetesPodPhase,
	},
	groupReplicaSets: []string {
		kubernetesReplicaSetAvailable,
		kubernetesReplicaSetDesired,
	},
	groupReplicationControllers: []string {
		kubernetesReplicationControllerAvailable,
		kubernetesReplicationControllerDesired,
	},
	groupResourceQuotas: []string {
		kubernetesResourceQuotaHard,
		kubernetesResourceQuotaUsed,
	},
}
```